### PR TITLE
Closes #143 Add RS dataset using PCWG3 criteria

### LINKED ISSUE #143: Feature Request: Add RS test data for prostate cancer
### Feature Idea

Create RS test data for prostate cancer.

### Relevant Input

_No response_

### Relevant Output

_No response_

### Reproducible Example/Pseudo Code

_No response_

### PR DESCRIPTION
**Pull request by tomyukn**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: data-raw/rs_onco_pcwg3.R [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+testnames <- tibble::tribble(
+  ~assessment,   ~RSTESTCD,                 ~RSTEST,                            ~RSCAT,
+  "PSA",          "TMRESP", "Tumor Marker Response", "PCWG SCHER PROSTATE CANCER 2016",
+  "SoftTissue", "SFTSRESP",  "Soft Tissue Response",                      "RECIST 1.1",
+  "Bone",       "BONERESP",         "Bone Response", "PCWG SCHER PROSTATE CANCER 2016",
+  "Overall",    "OVRLRESP",      "Overall Response", "PCWG SCHER PROSTATE CANCER 2016"
+)
+
+# date of the corresponding visit from LB
+data("lb")
```
Comment (Fanny-Gautier) [inline data-raw/rs_onco_pcwg3.R:60]: it is preferable to use 
lb <- pharmaversesdtm::lb

Comment (tomyukn) [reply data-raw/rs_onco_pcwg3.R]: Updated.

File: R/rs_onco_pcwg3.R [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+#' A SDTM RS dataset using PCWG3 criteria intended for examples of ADaM dataset creation
+#'
+#' @name rs_onco_pcwg3
+#' @title Disease Response (PCWG3)
+#' @keywords dataset
+#' @description A SDTM RS dataset using PCWG3 criteria intended for examples of ADaM dataset creation
+#' @docType data
+#' @format A data frame with 14 columns:
+#'   \describe{
+#'     \item{STUDYID}{undocumented field}
```
Comment (Fanny-Gautier) [inline R/rs_onco_pcwg3.R:15]: I believe that all variables should be properly labelled

File: R/rs_onco_pcwg3.R [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# This file is automatically generated by data-raw/create_sdtms_data.R.
+# For updating it please edit inst/extdata/sdtms-specs.json and rerun create_sdtms_data.R.
+# Manual edits are not recommended, as changes may be overwritten.
+#' Disease Response (PCWG3)
+#'
+#' A SDTM RS dataset using PCWG3 criteria intended for examples of ADaM dataset creation
```
Comment (Fanny-Gautier) [inline R/rs_onco_pcwg3.R:6]: Maybe add that it is for Oncology studies

File: R/rs_onco_pcwg3.R [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+#'
+#' @name rs_onco_pcwg3
+#' @title Disease Response (PCWG3)
+#' @keywords dataset
+#' @description A SDTM RS dataset using PCWG3 criteria intended for examples of ADaM dataset creation
```
Comment (Fanny-Gautier) [inline R/rs_onco_pcwg3.R:11]: Might be useful to add that it is for Oncology studies

File: inst/extdata/sdtms-specs.json [hunk 5]
```diff
@@ -175,6 +175,13 @@
       "source": "Generated dataset ([rs_supprs_onco_ca125.R](https://github.com/pharmaverse/pharmaversesdtm/blob/main/data-raw/rs_supprs_onco_ca125.R))",
       "author": "Vinh Nguyen"
     },
+    {
+      "name": "rs_onco_pcwg3",
+      "label": "Disease Response (PCWG3)",
+      "description": "A SDTM RS dataset using PCWG3 criteria intended for examples of ADaM dataset creation",
```
Comment (Fanny-Gautier) [inline inst/extdata/sdtms-specs.json:181]: Add that it is for Oncology studies

File: man/rs_onco_pcwg3.Rd [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+}
+}
+\source{
+Generated dataset
+}
+\usage{
+rs_onco_pcwg3
+}
+\description{
+A SDTM RS dataset using PCWG3 criteria intended for examples of ADaM dataset creation
```
Comment (Fanny-Gautier) [inline man/rs_onco_pcwg3.Rd:33]: for Oncology studies

File: man/rs_onco_pcwg3.Rd [hunk 7]
```diff
@@ -0,0 +1,43 @@
+% Generated by roxygen2: do not edit by hand
+% Please edit documentation in R/rs_onco_pcwg3.R
+\docType{data}
+\name{rs_onco_pcwg3}
+\alias{rs_onco_pcwg3}
+\title{Disease Response (PCWG3)}
+\format{
+A data frame with 14 columns:
+\describe{
+\item{STUDYID}{undocumented field}
```
Comment (Fanny-Gautier) [inline man/rs_onco_pcwg3.Rd:10]: add Label to all variables

**Review by Fanny-Gautier — CHANGES_REQUESTED** [review]
Minor comments to implement. Thanks

**Review by Fanny-Gautier — APPROVED** [review]
Looks good. Thanks for the updates!

**Comment by manciniedoardo** [comment]
@bundfussr are you happy for this to be merged, once @tomyukn fixes the WORDLIST conflicts?

**Comment by manciniedoardo** [comment]
Hi @bundfussr how is this looking? Keen to release pharmaversesdtm end of next week.

**Comment by bundfussr** [comment]
> Hi @bundfussr how is this looking? Keen to release pharmaversesdtm end of next week.

Hi @manciniedoardo , we have a meeting on Friday where we will discuss the new vignette, which uses the test data. I can let you know thereafter if the data is ready to go. If we need to make any updates, would it be OK to move the release?

**Comment by manciniedoardo** [comment]
I've moved it back another week so hopefully that's enough time!

**Comment by bundfussr** [comment]
Hi @tomyukn , could you add a case where there is only one assessment with PDu for combined response?

**Comment by tomyukn** [comment]
> Hi @tomyukn , could you add a case where there is only one assessment with PDu for combined response?

@bundfussr I have added a PDu only case.

**Comment by bundfussr** [comment]
> > Hi @tomyukn , could you add a case where there is only one assessment with PDu for combined response?
> 
> @bundfussr I have added a PDu only case.

Thanks

**Comment by bundfussr** [comment]
> @bundfussr are we good to go here?

I'm waiting for feedback from the admiralonco team. When do you need to know?

**Comment by bundfussr** [comment]
> I'd like to release Friday please @bundfussr as we need pharmaversesdtm to release for admiralophtha and admiralmetabolic

The admiralonco team is aware of that. Is it OK to merge on Friday or do you want to do it sooner?

**Comment by manciniedoardo** [comment]
The plan was to CRAN on Friday EU morning as @Lina2689 (india timezone) and @Fanny-Gautier will assist me in the process. So ideally we'd have this merged before then.

**Comment by bundfussr** [comment]
> The plan was to CRAN on Friday EU morning as @Lina2689 (india timezone) and @Fanny-Gautier will assist me in the process. So ideally we'd have this merged before then.

I see. Tomorrow afternoon?

**Comment by manciniedoardo** [comment]
> > The plan was to CRAN on Friday EU morning as @Lina2689 (india timezone) and @Fanny-Gautier will assist me in the process. So ideally we'd have this merged before then.
> 
> I see. Tomorrow afternoon?

Sounds good!

**Review by bundfussr — APPROVED** [review]
